modules/library/http/src/main/java/org/geotools/http/SimpleHttpClient.java

```diff
@@ -171,4 +196,36 @@ private URLConnection openConnection(URL finalURL, Map<String, String> headers)
         }
         return connection;
     }
+
+    private static boolean hasRedirect(int responseCode) {
+        return responseCode == HttpURLConnection.HTTP_SEE_OTHER
+                || responseCode == HttpURLConnection.HTTP_MOVED_PERM
+                || responseCode == HttpURLConnection.HTTP_MOVED_TEMP;
+    }
+
+    private URLConnection followRedirect(
+            HttpURLConnection connection, Map<String, String> headers, int redirectionCount)
+            throws IOException {
+        String redirect = connection.getHeaderField("Location");
+        if (redirect == null) {
+            LOGGER.warning("Tried to follow redirect but no url was provided in Location header");
+        } else if (redirectionCount < MAX_FOLLOW_REDIRECT) {
+            redirectionCount++;
+            try {
+                URL redirectURL = new URI(redirect).toURL();
+                LOGGER.info("Following redirect to " + redirect);
```

Comment:
This log level seems to high, it's a detail, but it would show up, for example, in the GeoServer default log level (which logs INFO and above).